app/src/main/java/de/westnordost/streetcomplete/tangram/QuestsMapFragment.java

```diff
@@ -176,12 +178,23 @@ private void zoomAndMoveToContain(ElementGeometry g)
 
 	private LngLat getCenterWithOffset(ElementGeometry geometry)
 	{
-		LngLat top = controller.screenPositionToLngLat(new PointF(0, questTopOffset));
-		LngLat bottom = controller.screenPositionToLngLat(new PointF(0, questBottomOffset));
-		LngLat offset = new LngLat(0,0);
-		offset.latitude = (top.latitude - bottom.latitude) / 2;
-		offset.longitude = (top.longitude - bottom.longitude) / 2;
 		LngLat pos = TangramConst.toLngLat(geometry.center);
+		LngLat offset = new LngLat(0,0);
+
+		if (getResources().getConfiguration().orientation == Configuration.ORIENTATION_PORTRAIT)
+		{
+			LngLat top = controller.screenPositionToLngLat(new PointF(0, questOffset.top));
+			LngLat bottom = controller.screenPositionToLngLat(new PointF(0, questOffset.bottom));
+			offset.latitude = (top.latitude - bottom.latitude) / 2;
+			offset.longitude = (top.longitude - bottom.longitude) / 2;
+		}
+		else if (getResources().getConfiguration().orientation == Configuration.ORIENTATION_LANDSCAPE)
+		{
+			LngLat left = controller.screenPositionToLngLat(new PointF(questOffset.left, 0));
+			LngLat right = controller.screenPositionToLngLat(new PointF(questOffset.right, 0));
+			offset.latitude = (left.latitude - right.latitude) / 2;
+			offset.longitude = (left.longitude - right.longitude) / 2;
+		}
```

Comment:
I think you did not understand the logic of this method.
This should work:
```
LngLat min = controller.screenPositionToLngLat(new PointF(questOffset.left, questOffset.bottom));
LngLat max = controller.screenPositionToLngLat(new PointF(questOffset.right, questOffset.top));
LngLat offset = new LngLat(0,0);
offset.latitude = (max.latitude - min.latitude) / 2;
offset.longitude = (max.longitude - min.longitude) / 2;
```

Reply:
Thank you!

Reply:
Tried to implement this, but the map centers without applying the offset (in portrait and landscape)...

Reply:
Time for a breakpoint. I don't claim to write 100% correct code on first try in a comment field.